CMakeLists.txt

```diff
@@ -10,6 +10,17 @@ project(CpuFeatures VERSION 0.7.0 LANGUAGES C)
 
 set(CMAKE_C_STANDARD 99)
 
+# Determine if cpu_features is built as a subproject (using add_subdirectory)
+# or if it is the master project.
+if (NOT DEFINED CPU_FEATURES_MAIN_PROJECT)
```

Comment:
@Mizux, should we add explanation of `CPU_FEATURES_MAIN_PROJECT` parameter to README?

Reply:
I would say no because it is just an internal temp variable result of the test `if(CMAKE_SOURCE_DIR STREQUAL PROJECT_SOURCE_DIR)`, so there is not intention to expose it to the user.

Concerning implementation I prefer the abseil-cpp one
https://github.com/abseil/abseil-cpp/blob/7e8d8018f621e94182876535320718542a4c5f09/CMakeLists.txt#L60-L66
while the option is duplicated it doesn't introduce a new `set()` variable...

```cmake
# when cpu_features is included as subproject (i.e. using add_subdirectory(cpu_features))
# in the source tree of a project that uses it, test rules are disabled.
if(NOT CMAKE_SOURCE_DIR STREQUAL PROJECT_SOURCE_DIR)
  option(BUILD_TESTING "Enable test rule" OFF)
else()
  option(BUILD_TESTING "Enable test rule" ON)
endif()
```

Reply:
@damageboy do you mind using @Mizux suggestion?

Reply:
Sorry, I totally missed this, I will fix according to @Mizux comment